Remove key from session by using session.delete

barbolo opened the pull request:
This PR only changes the guides.

Since you are not deleting a key from session when you assign nil to that key (`session[key] = nil`), the correct way of deleting a key from session should be `session.delete(key)`.

barbolo committed:
Remove key from session by using session.delete

You are not deleting a key from session when you assign nil to that key.

gmcgibbon commented on guides/source/action_controller_overview.md:
```diff
@@ -457,7 +457,7 @@ class LoginsController < ApplicationController
   def destroy
     # Remove the user id from the session
-    @_current_user = session[:current_user_id] = nil
+    session.delete(:current_user_id)
```
This would not clear the memoized user in `@_current_user` and could lead to unintended side effects. The user's ID is only retrieved from session once per request in this example so just deleting the session key wouldn't work.

barbolo committed:
Update guides on how to destroy a user session

In this commit, the user id is removed from session and controller's variables related to the user are nullified.
